metadataserver/identity.go

```diff
@@ -29,6 +31,39 @@ func (ms *MetadataServer) generateTaskIdentity() *titus.TaskIdentity {
 	}
 }
 
+// SetSigner updates the identity server's signer
+func (ms *MetadataServer) SetSigner(newSigner *identity.Signer) {
+	newCert, err := x509.ParseCertificate(newSigner.Certificate.Certificate[0])
+	if err != nil {
+		log.WithError(err).Error("Error parsing new certificate")
+	}
+
+	ms.signLock.Lock()
+	oldSigner := ms.signer
+	oldCert, err := x509.ParseCertificate(oldSigner.Certificate.Certificate[0])
+	if err != nil {
+		log.WithError(err).Error("Error parsing old certificate")
+	}
+
+	if newCert != nil && oldCert != nil {
```

Comment:
why not just do a blind 'return' in the `if err != nil` sections? or `x509.ParseCertificate` return errors?

Reply:
I got a little nervous about not doing the assignment when those parses fail.  I guess it's fine, though.